### PR TITLE
Switch to codecov.io

### PR DESCRIPTION
**Pull request by abhinav**

### CODE REVIEW

File: build/local.mk [hunk 1]
```diff
@@ -113,11 +113,12 @@ test: $(THRIFTRW) __eval_packages ## run all tests
...
-	go tool cover -html=cover.out -o cover.html
+	go tool cover -html=coverage.txt -o cover.html
 
-.PHONY: goveralls
-goveralls: cover $(GOVERALLS) ## run code coverage and upload to coveralls
-	PATH=$(BIN):$$PATH goveralls -coverprofile=cover.out -service=travis-ci
+.PHONY: codecov
+codecov: SHELL := /bin/bash
+codecov: cover ## run code coverage and upload to codecov.io
+	include_cov=coverage.txt bash <(curl -s https://codecov.io/bash)
```
Comment (willhug) [inline build/local.mk]: wat

File: .codecov.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+      experimental:
+        target: 70%
+        flags: experimental
+
+flags:
+
+  experimental:
+    paths:
+      - "/x/.*"
```
Comment (breerly) [inline .codecov.yml]: If this doesnt work, try explicitly listing out the paths, and we can follow up with the codecov peeps to add support for basic pattern matching.

File: build/local.mk (hunk 1 above)
Comment (breerly) [inline build/local.mk]: do you have a Q beyond what the comment describes?

I find the IoC here interesting :)

**Comment by breerly** [comment]
This is great - can you set `project/main` target to 81%?

File: scripts/cover.sh [hunk 3]
```diff
@@ -74,11 +74,16 @@ for pkg in "$@"; do
...
 done
 reset_waitpids
 
-gocovmerge "$COVER"/*.out > cover.out
+# Merge cross-package coverage and then split the result into main and
+# experimental coverages.
+gocovmerge "$COVER"/*.out \
+	| grep -v 'internal/examples' \
+	| tee >(grep -v /x/ > coverage.main.txt) \
+	| (echo 'mode: atomic'; grep /x/) > coverage.x.txt
```
Comment (abhinav) [inline scripts/cover.sh]: @prashantv approved this in-person

**Pull request merged**